GUIDELINES.md

```diff
@@ -207,6 +207,47 @@ Make sure all tests are passing with:
 
     $ cargo test --all-features
 
+### Running end-to-end tests
+
+In order to run end-to-end (e2e) tests you need to have a specific nightly toolchain.
+"Nightly" is necessary to use optimization compiler flags and have contract wasm small enough to be eligible for
+deployment.
+
+Run the following commands to install the necessary toolchain:
+
+```shell
+rustup install nightly-2024-01-01
+rustup component add rust-src
+```
+
+Also, you should have the cargo stylus tool:
+
+```shell
+cargo install cargo-stylus
+```
+
+Since most of the e2e tests using koba for deploying contracts, you need to install solidity compiler v0.8.21, like
```

Comment:
```suggestion
Since most of the e2e tests use [koba](https://github.com/OpenZeppelin/koba) for deploying contracts, you need to install the solidity compiler (`v0.8.21`). You can do so, following these steps:
```

Reply:
Do people need that exact version of the compiler? Or are we of with `0.8.21`+?

Reply:
I remember that koba doesn't work with latest solidity version

Reply:
It should work with up to `0.8.24` because that version introduces the `MCOPY` opcode, and `nitro-testnode` didn't support that at the time. [Issue](https://github.com/OpenZeppelin/koba/issues/4) from the koba repo.

Now that it does, the fix should be pretty simple, since I had already added `MCOPY` on [this commit](https://github.com/OpenZeppelin/koba/commit/607d36aae8ae43054d7215a1df7727ad3ad5f368). Probably only need to test with a newer Solidity version and that's it. However, there may be new stuff that nitro-testnode doesn't support.

cc: @bidzyyys @ggonzalez94